openssl_test.go

```diff
@@ -91,3 +92,24 @@ func compareCurrentVersion(v string) int {
 	ver := strings.TrimPrefix(runtime.Version(), "devel ")
 	return version.Compare(ver, v)
 }
+
+func TestFIPSProvider(t *testing.T) {
+	fipsProv := openssl.FIPSProvider()
+	if openssl.MajorVersion() == 1 {
+		want := openssl.FIPS()
+		if fipsProv != want {
+			t.Fatalf("FIPSProvider mismatch: want %v, got %v", want, fipsProv)
+		}
+		return
+	}
+
+	if openssl.FIPS() {
+		if !fipsProv {
+			t.Fatal("FIPSProvider should be true when FIPS is enabled")
+		}
+	} else if openssl.SymCryptProviderAvailable() {
+		if !fipsProv {
+			t.Fatal("FIPSProvider should be true when SymCrypt is available")
+		}
+	}
+}
```

Comment:
```go
func TestFIPSProvider(t *testing.T) {
	fipsProv := openssl.FIPSProvider()

	switch {
	case openssl.MajorVersion() == 1:
		if fipsProv != openssl.FIPS() {
			t.Fatalf("FIPSProvider mismatch: want %v, got %v", openssl.FIPS(), fipsProv)
		}
	case openssl.FIPS() || openssl.SymCryptProviderAvailable():
		if !fipsProv {
			t.Fatal("FIPSProvider should be true when FIPS or SymCrypt is enabled")
		}
	}
}
```
Since both required fipsProv to be true, This covers both scenarios where FIPS is enabled or SymCryptProviderAvailable is available. Also more informative error message. WDYT? (maybe you can choose if else over switch in this case depending on your style of programming)

Reply:
I thought adding switch case is good for declaring second case only triggers when openssl version is 3(kinda serving as a documentation). But if its safe to assume anything non major version 1 will be automatically get as 3 by readers with little or no context, it is an argument too.

Reply:
I see how the current implementation is somehow confusing. I've refactored the test to not depend on the OpenSSL version.